### PR TITLE
Added Group `refs`

### PR DESCRIPTION
**Pull request by RichardLitt**
Bugs:
- [ ] The `format` option has no effect on the output.
- [ ] If an argument is included, an `invalid ipfs paths ref` error is thrown,
  although there is no possible valid ipfs paths ref.

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -1036,6 +1036,332 @@ Outputs the list of peers that were removed.
...
+        Date: Fri, 05 Feb 2016 17:31:52 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes
+
+    + Body
+
+        ```
+        ```
```
Comment (dignifiedquire) [inline apiary.apib:5120]: Why add this section at all?

File: apiary.apib [hunk 2]
```diff
@@ -1036,6 +1036,332 @@ Outputs the list of peers that were removed.
...
+        Content-Type: application/json
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        X-Chunked-Output: 1
+        Date: Fri, 05 Feb 2016 17:31:52 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes
```
Comment (dignifiedquire) [inline apiary.apib]: Why add this section at all?

File: apiary.apib (hunk 1 above)
Comment (RichardLitt) [inline apiary.apib:5120]: To make it clear that an empty body is returned. How could this be done better?

File: apiary.apib (hunk 2 above)
Comment (RichardLitt) [inline apiary.apib]: See https://github.com/ipfs/http-api-spec/pull/37/files#r52057594

File: apiary.apib (hunk 1 above)
Comment (dignifiedquire) [inline apiary.apib:5120]: I would leave it out, but not sure

Comment (RichardLitt) [inline apiary.apib:5120]: I could add a line in the comment section about this, maybe?

File: apiary.apib [hunk 3]
```diff
@@ -1036,6 +1036,332 @@ Outputs the list of peers that were removed.
 
 # Group refs
 
+Retrieves the object named by <ipfs-path> and displays the link
+hashes it contains.
+
+## refs [GET /refs{?arg}{&format,edges,unique,recursive}]
+Lists links (references) from an object
```
Comment (dignifiedquire) [inline apiary.apib]: Missing `.`

File: apiary.apib [hunk 4]
```diff
@@ -1036,6 +1036,332 @@ Outputs the list of peers that were removed.
...
+
+#### curl
+    curl -i http://localhost:5001/api/v0/refs?arg=/ipfs/QmW2WQi7j6c7UgJTarActp7tDNikE4B2qXtFCfLPdsgaTQ/cat.jpg
```
Comment (dignifiedquire) [inline apiary.apib]: Wrong indentation / missing new line above

File: apiary.apib [hunk 5]
```diff
@@ -1036,6 +1036,332 @@ Outputs the list of peers that were removed.
...
+
++ Parameters
+    + arg (string, required) - Path to the object or objects to list refs from.
+    + format (string, optional) - Emit edges with given format. Tokens: `<src> <dst> <linkname>`.
```
Comment (dignifiedquire) [inline apiary.apib]: default value?

File: apiary.apib [hunk 6]
```diff
@@ -1036,6 +1036,332 @@ Outputs the list of peers that were removed.
...
+    + edges (boolean, optional) - Emit edge format: `<from> -> <to>`. Alias: e.
```
Comment (dignifiedquire) [inline apiary.apib]: default value?

File: apiary.apib [hunk 7]
```diff
@@ -1036,6 +1036,332 @@ Outputs the list of peers that were removed.
...
+    + unique (boolean, optional) - Omit duplicate refs from output. Alias: u.
```
Comment (dignifiedquire) [inline apiary.apib]: default value?

File: apiary.apib [hunk 8]
```diff
@@ -1036,6 +1036,332 @@ Outputs the list of peers that were removed.
...
+    + recursive (boolean, optional) - Recursively list links of child nodes. Alias: r.
```
Comment (dignifiedquire) [inline apiary.apib]: default value?

File: apiary.apib (hunk 1 above)
Comment (dignifiedquire) [inline apiary.apib:5120]: comment section?

Comment (RichardLitt) [inline apiary.apib:5120]: The optional description section for the response. I could just say, in instances like this, 'There is no body'

**Comment by RichardLitt** [comment]
In cases of default values: should I PR go-ipfs with a default value, too? @whyrusleeping @diasdavid

**Comment by daviddias** [comment]
> In cases of default values: should I PR go-ipfs with a default value, too? 

If there is a default value to be written on the API spec, I would assume that is already part of go-ipfs knowledge, however, if not, definitely PR or open a issue about it.

> - [ ] The format option has no effect on the output.

Because it is a CLI thing, not part of the HTTP API

> - [ ] If an argument is included, an invalid ipfs paths ref error is thrown,
>   although there is no possible valid ipfs paths ref.

Which kind of argument?

**Comment by daviddias** [comment]
As for the rest, LGTM.

Note: it would be really cool if we standardized on the IPFS Hashes format, sometimes it has to be multihash (/ipfs/HASH) others is just the hash, it can be annoying.

**Comment by RichardLitt** [comment]
Default values added here: https://github.com/ipfs/go-ipfs/pull/2582. For now, going to publish as is.

**Comment by RichardLitt** [comment]
Kind of argument: failed to replicate. Removed bug notice. Also removed format.

**Pull request merged**